manager/controlapi/node.go

```diff
@@ -234,16 +234,6 @@ func (s *Server) UpdateNode(ctx context.Context, request *api.UpdateNodeRequest)
 		return nil, grpc.Errorf(codes.NotFound, "node %s not found", request.NodeID)
 	}
 
-	if demote && s.raft != nil {
-		// TODO(abronan): the remove can potentially fail and leave the node with
-		// an incorrect role (worker rather than manager), we need to reconcile the
-		// memberlist with the desired state rather than attempting to remove the
-		// member once.
-		if err := s.raft.RemoveMember(ctx, member.RaftID); err != nil {
-			return nil, grpc.Errorf(codes.Internal, "cannot demote manager to worker: %v", err)
-		}
-	}
```

Comment:
I don't think we can move this part to `node`.

First, I believe this code needs to run on the leader, to prevent races.

Second, it means you can't remove an unresponsive member from the raft cluster, which is a big problem.

What I meant earlier about having the node shut down the manager was changing the raft code to not error out when it's removed from a cluster, and instead wait for the node code to call some function on the manager that shuts it down (based on a role change).